src/main/java/org/mule/extension/http/internal/request/HttpPollingSource.java

```diff
@@ -0,0 +1,186 @@
+
+  @Override
+  protected void doStart() throws MuleException {
+    LOGGER.error("Starting source");
+    scheduler = schedulerService.ioScheduler();
+    client = clientProvider.connect();
+    httpRequester = createHttpRequester(false, muleContext);
+    resolvedUri = getResolvedUri();
+  }
+
+  @Override
+  protected void doStop() {
+    LOGGER.error("Stopping source");
+    scheduler.stop();
+  }
+
+  private String getResolvedUri() {
+    UriParameters uriParameters = client.getDefaultUriParameters();
+    String resolvedPath = UriUtils.replaceUriParams(buildPath(config.getBasePath(), path), requestBuilder.getRequestUriParams());
+    return resolveUri(uriParameters.getScheme(), uriParameters.getHost().trim(), uriParameters.getPort(), resolvedPath);
+  }
+
+  protected String getId() {
+    return getClass().getSimpleName();
+  }
+
+  private void sendRequest(PollContext<String, HttpResponseAttributes> pollContext) {
+    CompletionCallback<InputStream, HttpResponseAttributes> callback =
+        new CompletionCallback<InputStream, HttpResponseAttributes>() {
+
+          @Override
+          public void success(Result<InputStream, HttpResponseAttributes> result) {
+            pollContext.accept(item -> {
+              // TODO (HTTPC-180): We put here splitting and etc... For now just consuming the stream
+              Result.Builder<String, HttpResponseAttributes> responseBuilder =
+                  Result.<String, HttpResponseAttributes>builder().output(IOUtils.toString(result.getOutput()));
+              result.getAttributes().ifPresent(attr -> responseBuilder.attributes(attr));
+              item.setResult(responseBuilder.build());
+            });
+          }
+
+          @Override
+          public void error(Throwable throwable) {
+            LOGGER.error("There was an error", throwable);
+          }
+        };
+
+    LOGGER.debug("Sending '{}' request to '{}'.", method, resolvedUri);
+    try {
+      httpRequester.doRequest(client, config, resolvedUri, method, config.getRequestStreamingMode(), config.getSendBodyMode(),
+                              config.getFollowRedirects(), client.getDefaultAuthentication(), config.getResponseTimeout(),
+                              responseValidator, transformationService, requestBuilder, true, muleContext, scheduler, null, null,
+                              callback, injectedHeaders, null);
+    } catch (MuleRuntimeException e) {
+      LOGGER.warn("Trigger '{}': Mule runtime exception found while executing poll: '{}'", getId(), e.getMessage(), e);
+    }
+
+  }
+
+  @Override
+  public void poll(PollContext<String, HttpResponseAttributes> pollContext) {
+    if (pollContext.isSourceStopping()) {
+      return;
+    }
+
+    sendRequest(pollContext);
+  }
+
+  @Override
+  public void onRejectedItem(Result<String, HttpResponseAttributes> result, SourceCallbackContext sourceCallbackContext) {
+    LOGGER.debug("Item rejected");
```

Comment:
should be info and contain more relevant information, like the correlation id or the id of the item being rejected. 

Also, if isDebugEnabled(), Result.getValue() should be logged as well.

Reply:
There is no correlation id at this point (we don't have the event to get it). Or do you mean from the response attributes?